### DISCUSSION TITLE
Modern Search v4 - April 2022

### DISCUSSION THREAD

**Discussion by FranckyC** [microsoft-search/pnp-modern-search#2000] [Announcements]
# PnP Modern Search v4

**This solution includes now only one `.sppkg` file:**
- `pnp-modern-search-parts-v4.sppkg`. Mandatory.

**Getting started documentation here:** [https://microsoft-search.github.io/pnp-modern-search/](https://microsoft-search.github.io/pnp-modern-search/)

[Feedback and comments](https://github.com/microsoft-search/pnp-modern-search/issues) are welcomed!

### New features

- PR #1931
     - Added a sort control component to layouts:
          - Now the list of sortable properties is a static list of 'Sortable' managed properties in the search schema. Custom properties or RefinableXXX have to be entered manually.
          - Updated the Details List sort behavior. Now there is no static sorting anymore. The field used to sort a column comes from the data source sort settings configuration and we clearly make the distinction between display value and sort value in the configuration.
- PR #1873 - Included new function for Handlebars Helper.
- PR #1989 - Added Spanish translations.
- PR #1987 - Added German translations.
- PR #1786 - Feature/fix query alteration options
- PR #1908 - Added pagination to the 'Custom' layout

### Fixes

- PR #1986 - Fixed #1372 Fix momentjs locale 
- PR #1980 - Fixed #1840 Added a conditional check for the data source reference property.
- PR #1979 - Fixed #1965 Added properties and utils from the main context to the Details List and Cards Handlebars context.
- PR #1952 - Fixed #1893 Added note in documentation about SPFx version for extensions.
- PR #1951 - Fixed #1943 Combo box search filter is now case insensitive.
- PR #1937 - Added data-instance-id to pnp-slider template.
- PR #1879 - Fixed #1878 Fix for multi-value UPA values.
- PR #1837 - Fixed #1797 Only enclosed term text value with the string() operator. Other values remain HEX encoded.
- PR #1836 - Fixed #1835. Refinement token shouldn't be enclosed with quotes if contains the range() FQL operator.

## Search filters

N/A

## Search Verticals

N/A

## Search Box

N/A

## Key contributors

- @wobba 
- @FranckyC 
- @gszdev
- @kachihro
- @edarroudi
- @eduardpaul
- @edarroudi 
- @dominiqueborer
- @kflogdev

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft-search/pnp-modern-search/releases/tag/4.6.0'>Modern Search v4 - April 2022</a>.</em>

**Comment by edarroudi** [comment]
Hey @FranckyC  - i might have found a problem.
I wanted to check my code regarding #1786 but the new solution wasn't showing correctly. To check if the new version is loaded correctly i looked at change #1908. But that wasn't showing correctly either. The 4.6.0 version is showing correctly. So i thought maybe a public cdn problem or something like that. But the files are identical. So i looked at the chunks and saw that https://github.com/microsoft-search/pnp-modern-search/blob/7710b8a5e9651dadb55f36400ed8c9f77c2f0f97/search-parts/src/dataSources/MicrosoftSearchDataSource.ts#L414 is not reflected correctly. It seems to be an old version.

See file chunk.pnp-modern-search-microsoft-search-datasource_9985eb76031933f9eb75.js
![image](https://user-images.githubusercontent.com/935493/161937883-865597ac-da70-42c5-bfdc-85d59ff425ea.png)

I can write more about it later - have to go to a meeting...

**Reply by edarroudi** [reply]
so yeah. I rebuilt it locally and the result is fine: 
![image](https://user-images.githubusercontent.com/935493/161970860-fa76e0a2-ec00-4d58-847d-79a73d26ec24.png)
So something went wrong in the build / bundle .

**Reply by FranckyC** [reply]
I updated the sppkg package. Can you try again?

**Reply by edarroudi** [reply]
Yeah. Works fine now. Thank :)

**Comment by edarroudi** [comment]
Hey @FranckyC ,
I did check the german language now and I have found a "bug" with moment and fixed it : #2006
Hope you can create a new hotfix version? Sorry for that, next time I do a full test.
